storage/innobase/log/log0recv.cc

```diff
@@ -3563,6 +3564,12 @@ inline buf_block_t *recv_sys_t::recover_low(const map::iterator &p, mtr_t &mtr,
     }
   }
 
+  /* In case of corruption, block will be unlocked inside
+  corrupted_evict() function. In case of successful recovery,
+  block lock will be released in recover_low() for
+  non-deferred tablespace. For deferred tablespace,
+  InnoDB releases the block in recover_deferred() */
+  block->page.lock.x_lock_recursive();
```

Comment:
There is no `recv_sys_t::corrupted_evict()`. Let us use the complete function `buf_pool_t::corrupted_evict()`. There are two `recv_sys_t::recover_low()`. I think that the following comment would be clearer:
```c++
  /* Released in buf_pool_t::corrupted_evict(), recover_deferred(), or below */
  block->page.lock.x_lock_recursive();
```